tests/unit/lib/utilities/filters/test_filters.py

```diff
@@ -0,0 +1,183 @@
+
+import unittest
+
+from liota.lib.utilities.filters.range_filter import RangeFilter, Type
+
+
+class RangeFilterTest(unittest.TestCase):
+    """
+    Range Filter Unit Test Cases
+    """
+    def setUp(self):
+        """Initialize lower and upper bound for filter test"""
+        self.lower_bound = 10
+        self.upper_bound = 20
+
+    def test_init_filter_type (self):
+        """If invalid filter type, raise TypeError"""
+        self.assertRaises(TypeError, lambda: RangeFilter(None, 0, 1))
+
+    def test_init_lower_upper_bound(self):
+        """If both lower and upper bound is not a number, raise TypeError"""
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.CLOSED, "0", "1"))
+
+    def test_init_lower_bound(self):
+        """If lower bound is not a number, raise TypeError"""
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.CLOSED, "0", 1))
+
+    def test_init_upper_bound(self):
+        """If upper bound is not a number, raise TypeError"""
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.CLOSED, 1, "2"))
+
+    def test_init_lower_less_than(self):
+        """In filter LESS_THAN if lower bound is not a number, raise TypeError"""
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.LESS_THAN, "0", 1))
+
+    def test_init_upper_greater_than(self):
+        """In filter GREATER_THAN if upper bound is not a number, raise TypeError"""
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.GREATER_THAN, 1, "2"))
+
+    """
+    Accept filters : Bounded at both end
+    Values : Ranging inside limit
+    """
+    def test_number_closed(self):
+        """If any filter applied on value which is not a number, value will be returned as it is"""
+        closed_filter = RangeFilter(Type.CLOSED, self.lower_bound, self.upper_bound)
+        value = closed_filter.filter("6")
+        self.assertEquals(value, "6")
+
+    def test_closed_filter(self):
+        """If CLOSED filter applied value will be accepted if: lower_bound <= value <= upper_bound"""
+        closed_filter = RangeFilter(Type.CLOSED, self.lower_bound, self.upper_bound)
+        value = closed_filter.filter(15)
+        self.assertEquals(value, 15)
+
+    def test_open_filter(self):
+        """If OPEN filter applied value will be accepted if: lower_bound < value < upper_bound"""
+        open_filter = RangeFilter(Type.OPEN, self.lower_bound, self.upper_bound)
+        out_range_value = open_filter.filter(10)
+        in_range_value = open_filter.filter(15)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 15)
+
+    def test_closed_open_filter(self):
+        """If CLOSED_OPEN filter applied value will be accepted if: lower_bound <= value < upper_bound """
+        closed_open_filter = RangeFilter(Type.CLOSED_OPEN, self.lower_bound, self.upper_bound)
+        out_range_value = closed_open_filter.filter(20)
```

Comment:
In this filter type, values will be filtered out if this condition is met: lower_bound <= value < upper_bound.  So, here lower bound itself is outside range.  Another test for outside range would be redundant.